clusterman/aws/aws_resource_group.py

```diff
@@ -140,6 +140,13 @@ def terminate_instances_by_id(self, instance_ids: List[str], batch_size: int = 5
 
         instance_weights = {}
         for instance in ec2_describe_instances(instance_ids):
+            # skip if instance already terminated
+            if instance["State"].get("Name") == "terminated" or instance["State"].get("Name") == "shutting-down":
```

Comment:
this can be changed with (if instance["State"].get("Name") in  {"terminated", "shutting-down"}